File: webapi/Models/HR/Income.cs
using System.ComponentModel.DataAnnotations;

namespace webapi.Models.HR
{
    public class Income
    {
        [Key]
        public int Id { get; set; }
        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public decimal HealtInsuranceArt40 { get; set; }
        public decimal TotalInsurance { get; set; }
        public decimal HealthInsurance { get; set; }
        public decimal GrossRemuneration { get; set; }
        public decimal BonusIncome { get; set; }
        public decimal AdditionalIncome { get; set; }
        public DateOnly CreationDate { get; set; }
    }


    public class IncomeDTO
    {
        public int EmployeeId { get; set; }
        public decimal HealtInsuranceArt40 { get; set; }
        public decimal TotalInsurance { get; set; }
        public decimal HealthInsurance { get; set; }
        public decimal GrossRemuneration { get; set; }
        public decimal BonusIncome { get; set; }
        public decimal AdditionalIncome { get; set; }
    }
}

File: reactapp/src/pages/EndMonth/models/Income.ts
export interface Income {
    id: int,
    employeeId: int,
    healtInsuranceArt40: string,
    totalInsurance: string,
    healthInsurance: string
    grossRemuneration: string,
    bonusIncome: string,
    additionalIncome: string,
    creationDate: string
}


export interface IncomeInsert {
    employeeId: int,
    healtInsuranceArt40: string,
    totalInsurance: string,
    healthInsurance: string
    grossRemuneration: string,
    bonusIncome: string,
    additionalIncome: string,
}


export const createDefaultIncomeInsert = (employeeId: int): IncomeInsert => {
    return {
        employeeId: employeeId,
        healtInsuranceArt40: "0.00",
        totalInsurance: "0.00",
        healthInsurance: "0.00",
        grossRemuneration: "0.00",
        bonusIncome: "0",
        additionalIncome: "0",
    }
}



export const defaultIncomeUpdate: Income = {
    id: 0,
    employeeId: 0,
    healtInsuranceArt40: "",
    totalInsurance: "",
    healthInsurance: "",
    grossRemuneration: "",
    bonusIncome: "",
    additionalIncome: "",
    creationDate: ""
}
